### DISCUSSION TITLE
Help with import templates syntax

### DISCUSSION THREAD

**Discussion by richard-lee-863** [ananthakumaran/paisa#125] [Q&A]
There may need to be a bit more clarity in the Docs about when and when not to use certain syntax, such as the wrapping {{ }}. 
Especially for those new to Handlebars. 

I've had a look through Handlebars documentation as well, but things seem to not be super clear. 

For instance, why does this:
`{{date ROW.A "DD/MM/YYYY"}} {{ROW.C}}`
Have {{ }} around the entire date function, and why does the date function not have normal brackets () like you'd expect? 

And why do some functions seem to not need brackets,, where others do? 
For instance:
` {{trim ROW.D}}`
Couldn't that be written as 
`trim({{ROW.D}})`

I have a few syntax issues to work through which i'll mention below:

**Comment by richard-lee-863** [comment]
Inline function use with mixed fixed text and functions.
Trying to round very large decimal place buy orders so that the double accounting math works out nicely. 

`Assets:Equity:AIR	{{round (ROW.I, 4)}} AirNZ @ {{ROW.J}} NZD`

![image](https://github.com/ananthakumaran/paisa/assets/43104957/097ca21f-a6a0-4f29-b269-8b7e72e6f047)

EDIT: 

How are we even supposed to deal with rounding properly? I notice that my raw data copied into a CSV file has more decimal places that what Paisa is parsing into the "preview pane". 
for example 423.0000024 is shown in Paisa, but it is actually 423.00000239 in my CSV.
So all entries should already be balanced but Paisa is complaining about tiny differences that result from the way it itself is importing the data..?

**Reply by ananthakumaran** [reply]
@richard-lee-863

 for these kinds of rapid questions matrix chat would be better https://matrix.to/#/#paisa:matrix.org, we can also help with the import template if you share a sample CSV file. The import template syntax might not be intuitive if you are not used to programming.

anything outside {{}} is plain text, not executed, anything inside {{}} is code. try `{{round ROW.I precision=2}}` for rounding. 

> for example 423.0000024 is shown in Paisa, but it is actually 423.00000239 in my CSV.

This is likely the importer issue as it uses javascript float for storing numbers, which has limited precision.

**Reply by richard-lee-863** [reply]
Thanks I will try use matrix more if that is preferred. 

Hmm limited precision will not be good for trying to track things like crypto, when typically you might purchase very small amounts (of Bitcoin for example).

**Reply by ananthakumaran** [reply]
Understood how it could become an issue. Only the importer has this problem because it uses the javascript env. I will check if it would be possible to use BigNumber in the importer as well.
